src/corrgi/correlation/correlation.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Callable
+
+import numpy as np
+import pandas as pd
+from hipscat.catalog.catalog_info import CatalogInfo
+from munch import Munch
+
+from corrgi.utils import project_coordinates
+
+
+class Correlation(ABC):
+    """Correlation base class."""
+
+    def __init__(
+        self,
+        left_df: pd.DataFrame,
+        right_df: pd.DataFrame | None,
+        left_catalog_info: CatalogInfo,
+        right_catalog_info: CatalogInfo | None,
```

Comment:
Could you move these to be method params so that you don't have the None for auto?

Reply:
Thank you, that's an excellent idea!